pkg/operator/ceph/pool/controller_test.go

```diff
@@ -618,17 +619,71 @@ func TestConfigureRBDStats(t *testing.T) {
 	poolWithRBDStatsEnabled.Spec.EnableRBDStats = true
 	objects = append(objects, poolWithRBDStatsEnabled)
 	context.Client = fake.NewClientBuilder().WithScheme(s).WithRuntimeObjects(objects...).Build()
-	err = configureRBDStats(context, clusterInfo)
+	err = configureRBDStats(context, clusterInfo, "")
 	assert.Nil(t, err)
 
-	// Case 5: Two CephBlockPools with EnableRBDStats:false & EnableRBDStats:true.
+	// Case 5: Two external pools(non CephBlockPools) with rbd_stats_pools config already set
+	monStore := config.GetMonStore(context, clusterInfo)
+	e := monStore.Set("mgr", "mgr/prometheus/rbd_stats_pools", "pool1,pool2")
+	assert.Nil(t, e)
+	context.Executor = &exectest.MockExecutor{
+		MockExecuteCommandWithTimeout: func(timeout time.Duration, command string, args ...string) (string, error) {
+			logger.Infof("Command: %s %v", command, args)
+			if args[0] == "config" && args[2] == "mgr" && args[3] == "mgr/prometheus/rbd_stats_pools" {
+				if args[1] == "set" && args[4] != "" {
+					return "", nil
+				}
+				if args[1] == "get" {
+					return "my-pool-with-rbd-stats,pool1,pool2", nil
+				}
+				if args[1] == "rm" {
+					return "", nil
+				}
+			}
+			return "", errors.Errorf("unexpected arguments %q", args)
+		},
+	}
+	e = configureRBDStats(context, clusterInfo, "")
+	assert.Nil(t, e)
+
+	rbdStatsPools, err := monStore.Get("mgr", "mgr/prometheus/rbd_stats_pools")
+	assert.Nil(t, err)
+	assert.Equal(t, "my-pool-with-rbd-stats,pool1,pool2", rbdStatsPools)
+
+	// Case 6: Deleted CephBlockPool should be excluded from config
+	context.Executor = &exectest.MockExecutor{
+		MockExecuteCommandWithTimeout: func(timeout time.Duration, command string, args ...string) (string, error) {
+			logger.Infof("Command: %s %v", command, args)
+			if args[0] == "config" && args[2] == "mgr" && args[3] == "mgr/prometheus/rbd_stats_pools" {
+				if args[1] == "set" && args[4] != "" {
+					return "", nil
+				}
+				if args[1] == "get" {
+					return "pool1,pool2", nil
```

Comment:
Instead of duplicating this whole MockExecutor, how about a variable in the mock so you can just change what it returns for each test, something like this:
```suggestion
					return mockedPools, nil
```